lib/controllers.js

```diff
@@ -1,9 +1,16 @@
 'use strict';
 
+var groups = require.main.require('./src/groups');
+
 var Controllers = {};
 
 Controllers.renderAdminPage = function (req, res) {
-	res.render('admin/plugins/session-sharing', {});
+	groups.getGroupsFromSet('groups:visible:createtime', req.uid, 0, -1, function(err, groupData) {
+		if (err) {
+			return next(err);
+		}
+		res.render('admin/plugins/session-sharing', { groups: groupData });
+	});
```

Comment:
For fun, you can now do:

```
var groups = require.main.require('./src/groups').async;

var groupData = await groups.getGroupsFromSet('groups:visible:createtime', req.uid, 0, -1);
res.render('admin/plugins/session-sharing', { groups: groupData });
```

:smiling_imp:

Reply:
How would an error from fetching groups be handled with async await? Would the error be assigned to groupData?

Reply:
Also is this specific to a certain nodebb version or just using latest JS techniques? As you can tell I haven't had chance to play with native async workflows

Reply:
It's new as of 1.10.2 I think... ES6 can be used on the server-side (though not client-side as we're still aiming for IE11 compatibility)

Reply:
Oh, and error handling is via try..catch... which we don't really do yet.